src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -32,7 +32,7 @@
     @Query("select distinct course from Course course left join fetch course.exercises exercises left join fetch course.lectures lectures left join fetch lectures.attachments left join fetch exercises.categories where (course.startDate <= current_timestamp or course.startDate is null) and (course.endDate >= current_timestamp or course.endDate is null)")
     List<Course> findAllActiveWithEagerExercisesAndLectures();
 
-    @EntityGraph(type = LOAD, attributePaths = { "exercises" })
+    @EntityGraph(type = LOAD, attributePaths = { "exercises", "exercises.categories", "exercises.teamAssignmentConfig" })
     Course findWithEagerExercisesById(long courseId);
```

Comment:
I'm not sure if this is relevant, but the method name no longer indicates exactly what is loaded.

Reply:
Hm yeah, we need some kind of convention for this. I'm not sure if it's resonable to name all the loaded attributes in the method name when there are a lot of them, e.g. in this case `findWithEagerExercisesAndExerciseCategoriesAndExerciseTeamAssignmentConfigById`.

Since `categories` and `teamAssignmentConfig` are nested on the `exercises` attributes, maybe we could say it's fine to omit them in the method name?

This would provide a reasonable answer in this case but it still wouldn't tell us what to do in other cases like the one you mentioned below where there are a lot of attributes on the top level:

```java
@EntityGraph(type = LOAD, attributePaths = { "exampleSubmissions", "teamAssignmentConfig", "categories" })
    Optional<ModelingExercise> findWithEagerExampleSubmissionsById(@Param("exerciseId") Long exerciseId);
```

The spring documentation offers no guidance here. They do not include loaded attributes in the method name in any way:
![OnPaste 20200425-150254](https://user-images.githubusercontent.com/7109225/80280684-e385bb80-8705-11ea-9a4c-5319955fb47d.png)

But I don't think that is something that we could do since we might have repository methods that only differ in the attributes being loaded, and thus they need a different name to be distinguishable.

We could say that we do not include eagerly loaded attributes in the repository method name in general but only if it is important for distinction or if it is a very expensive eager load. This convention wouldn't be super strict since those points are debatable on a case by case basis but probably better than no convention at all. Or maybe somebody else has a better idea altogether?

Reply:
This is actually very closely related to what @roxhens has been working on in [this](https://github.com/ls1intum/Artemis/pull/1331) PR. Maybe he has some thoughts on this?